cpp/src/parquet/file_reader.h

```diff
@@ -58,6 +59,11 @@ class PARQUET_EXPORT RowGroupReader {
   // column. Ownership is shared with the RowGroupReader.
   std::shared_ptr<ColumnReader> Column(int i);
 
+  // Construct a RecordReader for the indicated row group-relative column i.
+  // Ownership is shared with the RowGroupReader.
+  std::shared_ptr<internal::RecordReader> RecordReader(
+      int i, bool read_dictionary = false, bool read_dense_for_nullable = false);
```

Comment:
Can we wrap `read_dictionary` and `read_dense_for_nullable` into ArrowReaderProperties / ReaderProperties? These properties are easily expanded to make it difficult to maintain. Both ArrowReaderProperties and ReaderProperties are available in the RowGroupReader, so we can simplify the interface to `std::shared_ptr<RecordReader> RecordReader(int i)`.

Reply:
Done.
It turns out that read_dictionary is a redundant argument for RecordReader::Make and does not do anything.